src/pages/index.js

```diff
@@ -47,13 +73,165 @@ const cardListEl = new Section(
+function handleSubmit(request, popupInstance, loadingText = "Saving...") {
+  // here we change the button text
+  popupInstance.renderLoading(true, loadingText);
+  request()
+    .then(() => {
+      // We need to close only in 'then'
+      popupInstance.close();
+      // Disable submit button only in a 'then' block after a successful response to be able to click it again if there is a server error.
+      popupInstance.resetValidation();
+    })
+    // we need to catch possible errors
+    .catch(console.error)
+    // in 'finally' return initial button text back in any case
+    .finally(() => {
+      popupInstance.renderLoading(false);
+    });
+}
+
+// Like and unlike
+function handleLikeIcon(card) {
+  if (!card.isLiked) {
+    // if card is not liked, like it
+    api
+      .likeCard(card._id) //card._id is the id of the card
+      .then(() => {
+        console.log(card);
+        //shouldn't set value of isLiked outside the card. this is now set in setButtonState()
+        // card.isLiked = true;
+        card.setIsLiked(true);
+      })
+      .catch(console.error);
+  } else {
+    api
+      .unlikeCard(card._id) // card._id is the id of the card
+      .then(() => {
+        //shouldn't set value of isLiked outside the card. this is now set in setButtonState()
+        // card.isLiked = false;
+        card.setIsLiked(false);
+      })
+      .catch(console.error);
+  }
+}
+
+// // Like and unlike
+// function handleLikeIcon(card) {
+//   if (!card._isLiked) {
+//     // if card is not liked, like it
+//     api
+//       .likeCard(card) //card._id is the id of the card
+//       .then(() => {
+//         console.log(cardID);
+//         console.log(card);
+//         card.isLiked = true;
+//         card.setButtonState();
+//       })
+//       .catch(console.error);
+//   } else {
+//     api
+//       .unlikeCard(cardID) // card._id is the id of the card
+//       .then(() => {
+//         card.isLiked = false;
+//         card.setButtonState();
+//       })
+//       .catch(console.error);
+//   }
+// }
+
+// Delete modal and methods
+const deleteCardPopup = new PopupWithConfirm({
+  popupSelector: "#delete-confirm-modal",
 });
+deleteCardPopup.setEventListeners();
+
+function handleDeleteCard(cardID, card) {
+  deleteCardPopup.open();
+  deleteCardPopup.handleDeleteConfirm(() => {
+    deleteCardPopup.renderLoading(true);
+    api
+      .deleteCard(cardID)
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardPopup.close();
+      })
+      .catch(console.error)
+      .finally(() => {
+        deleteCardPopup.renderLoading(false);
+      });
+  });
+}
+
+// AVATAR MODAL AND METHODS
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarFormSubmit,
+  avatarFormValidator
+);
+editAvatarPopup.setEventListeners();
+
+editAvatarButton.addEventListener("click", () => {
+  editAvatarPopup.open();
+  avatarFormValidator.resetValidation();
```

Comment:
`  avatarFormValidator.resetValidation();
`

It's not needed here because you do that inside `handleSubmit`